### PR TITLE
Updated documentation for Grammar Backend

### PR DESCRIPTION
**Pull request by shuaills**
## Motivation

Improve documentation clarity around grammar backends.

https://github.com/sgl-project/sglang/issues/2505

## Modifications

Updated documentation to clarify Xgrammar's JSON support and regex limitation.

## Checklist
- [x] Format your code according to the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/references/contributor_guide.md).
- [x] Add unit tests as outlined in the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/references/contributor_guide.md).
- [x] Update documentation as needed, including docstrings or example tutorials.

### CODE REVIEW

File: docs/backend/openai_api_completions.ipynb [hunk 1]
```diff
@@ -223,6 +223,13 @@
     "## Structured decoding (JSON, Regex)\n",
```
Comment (zhaochenyang20) [inline docs/backend/openai_api_completions.ipynb:223]: You can define a JSON schema or regular expression to constrain the model's output, which depends on the grammar backend.

SGlang has two backends: outlines (default) and Xgrammar. Xgrammar enhances JSON decoding performance but does not support regular expressions. To use Xgrammar, add --grammar-backend xgrammar when launching the server:

**Review by zhaochenyang20 — CHANGES_REQUESTED** [review]

**Comment by zhaochenyang20** [comment]
This is just to update the current document, so a tiny modification is good enough. However, for the constraint decoding tutorial, please truly launch the server to check its correctness. Also, what's the status of the tutorial? @shuaills

File: docs/backend/openai_api_completions.ipynb (hunk 1 above)
Comment (shuaills) [reply docs/backend/openai_api_completions.ipynb:223]: Updated and verified the functionality with Xgrammar backend.

**Comment by shuaills** [comment]
> This is just to update the current document, so a tiny modification is good enough. However, for the constraint decoding tutorial, please truly launch the server to check its correctness. Also, what's the status of the tutorial? @shuaills

I'm working on it. Should it be in Jupyter Notebook format? And where should I put the tutorial?

**Comment by zhaochenyang20** [comment]
> > This is just to update the current document, so a tiny modification is good enough. However, for the constraint decoding tutorial, please truly launch the server to check its correctness. Also, what's the status of the tutorial? @shuaills
> 
> I'm working on it. Should it be in Jupyter Notebook format? And where should I put the tutorial?

It should be a Notebook. Put it under `docs/backend`

File: docs/backend/openai_api_completions.ipynb [hunk 2]
```diff
@@ -221,7 +221,14 @@
    "metadata": {},
    "source": [
     "## Structured decoding (JSON, Regex)\n",
-    "You can specify a JSON schema or a regular expression to constrain the model output. The model output will be guaranteed to follow the given constraints.\n",
+    "You can define a JSON schema or regular expression to constrain the model's output, which depends on the grammar backend.\n",
```
Comment (zhaochenyang20) [inline docs/backend/openai_api_completions.ipynb]: You can define a JSON schema or regular expression to constrain the model's output. The model output will be guaranteed to follow the given constraints and this depends on the grammar backend.

**Review by zhaochenyang20 — CHANGES_REQUESTED** [review]
Just tiny modification. Then we can merge it.

File: docs/backend/openai_api_completions.ipynb (hunk 2 above)
Comment (shuaills) [reply docs/backend/openai_api_completions.ipynb]: done, thanks

**Review by zhaochenyang20 — APPROVED** [review]

**Pull request merged**